Plugin fails to enable

### Server Logs

[19:13:17] [Server thread/INFO]: [AntiRedstoneClock-Remastered] Enabling AntiRedstoneClock-Remastered v2.2.4
[19:13:17] [Server thread/INFO]: [net.onelitefeather.antiredstoneclockremastered.injection.ServiceModule] Using modern translation service
[19:13:17] [Server thread/INFO]: [net.onelitefeather.antiredstoneclockremastered.service.factory.RedstoneClockServiceFactory] Using BukkitRedstoneClockService
[19:13:17] [Server thread/ERROR]: Error occurred while enabling AntiRedstoneClock-Remastered v2.2.4 (Is it up to date?)
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) [Guice/ErrorInCustomProvider]: IllegalStateException: New flags cannot be registered at this time
  at ExternalSupportModule.provideWorldGuardSupport(ExternalSupportModule.java:43)
  while locating WorldGuardSupport

Learn more:
  https://github.com/google/guice/wiki/ERROR_IN_CUSTOM_PROVIDER

1 error

======================
Full classname legend:
======================
ExternalSupportModule: "net.onelitefeather.antiredstoneclockremastered.injection.ExternalSupportModule"
WorldGuardSupport:     "net.onelitefeather.antiredstoneclockremastered.api.WorldGuardSupport"
========================
End of classname legend:
========================

	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InternalProvisionException.toProvisionException(InternalProvisionException.java:251) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1151) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1186) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/net.onelitefeather.antiredstoneclockremastered.AntiRedstoneClockRemastered.onEnable(AntiRedstoneClockRemastered.java:93) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:280) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:202) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:109) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:520) ~[paper-api-1.21.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.CraftServer.enablePlugin(CraftServer.java:658) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at org.bukkit.craftbukkit.CraftServer.enablePlugins(CraftServer.java:607) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:743) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:488) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:322) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1163) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:310) ~[paper-1.21.4.jar:1.21.4-232-12d8fe0]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.IllegalStateException: New flags cannot be registered at this time
	at worldguard-bukkit-7.0.13-dist.jar/com.sk89q.worldguard.protection.flags.registry.SimpleFlagRegistry.register(SimpleFlagRegistry.java:60) ~[worldguard-bukkit-7.0.13-dist.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/net.onelitefeather.antiredstoneclockremastered.worldguard.v7.WorldGuardModernSupport.registerFlag(WorldGuardModernSupport.java:94) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/net.onelitefeather.antiredstoneclockremastered.injection.ExternalSupportModule.provideWorldGuardSupport(ExternalSupportModule.java:59) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/net.onelitefeather.antiredstoneclockremastered.injection.ExternalSupportModule$$FastClassByGuice$$1bffdb.GUICE$TRAMPOLINE(<generated>) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/net.onelitefeather.antiredstoneclockremastered.injection.ExternalSupportModule$$FastClassByGuice$$1bffdb.apply(<generated>) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:260) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.ProviderMethod.doProvision(ProviderMethod.java:171) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.provision(InternalProviderInstanceBindingImpl.java:185) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.get(InternalProviderInstanceBindingImpl.java:162) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:169) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:45) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	at AntiRedstoneClock-Remastered-2.2.4.jar/com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1148) ~[AntiRedstoneClock-Remastered-2.2.4.jar:?]
	... 14 more
[19:13:17] [Server thread/INFO]: [AntiRedstoneClock-Remastered] Disabling AntiRedstoneClock-Remastered v2.2.4

### Expected behavior

AntiRedstoneClock-Remastered should enable successfully and register its WorldGuard flags without errors.


### Actual behavior

The plugin fails to enable on server startup. It prints an IllegalStateException about new flags not being able to register.


### Steps to reproduce

1. Run a Paper 1.21.4 server with WorldGuard 7.0.13.
2. Install AntiRedstoneClock-Remastered v2.2.4.
3. Start the server.
4. Watch console — plugin fails to enable.

### Plugin and Datapack List

pl
[19:43:39 INFO]: ℹ Server Plugins (107):
[19:43:39 INFO]: Paper Plugins (7):
[19:43:39 INFO]:  - AntiRedstoneClock-Remastered, EconomyShopGUI, ExcellentEnchants, nightcore, SignedVelocity, SmartSpawner, TotemGuard
[19:43:39 INFO]: Bukkit Plugins (100):
[19:43:39 INFO]:  - AdvancedBanX, AdvancedBanX-MutePreventions, AdvancedPlHide, AdvancedReplay, *AdvancedRestart, AdvancedSpyInventory, ajLeaderboards, ajQueue, AltDetector, AntiAutoFishing
[19:43:39 INFO]:  AntiCheatReplay, AuctionHouse, AuraSkills, Aurora, AuroraCollections, AuroraLevels, AutoFarmRegions, AxAFKZone, AxMinions, AxPlayerWarps
[19:43:39 INFO]:  AxTrade, AxVaults, BentoBox, BentoBox-AOneBlock, BentoBox-Border, BentoBox-Level, ChatFeelings, ChatFilter, ChatGames-Premium, Citizens
[19:43:39 INFO]:  ClearLag, CommandAPI, *CommandSpy, ControlPlayer, CoreProtect, CrazyEnchantments, CustomJoinMessages, CustomOreGen, DamagerENW, DecentHolograms
[19:43:39 INFO]:  DeluxeCoinflip, DeluxeMenus, DeluxeTags, DiscordSRV, ElytraBlocker, Essentials, EssentialsChat, EssentialsSpawn, EvenMoreFish, ExcellentCrates
[19:43:39 INFO]:  EZColors, fakeplayer, FancyGlow, FastAsyncWorldEdit, FreedomChat, GSit, HarvestEXP, Hitmarkers, Infiniteannouncements, InteractiveChat
[19:43:39 INFO]:  InventoryRollbackPlus, ItemEdit, ItemsAdder, KnockbackSync, LibsDisguises, LoneLibs, LuckPerms, Maintenance, ModernChatInjector, Multiverse-Core
[19:43:39 INFO]:  NBTAPI, NoteBlockAPI, packetevents, PlaceholderAPI, PlayerCrasher, PlayerPoints, *PlayTime, ProtocolLib, PvPManager, QuickShop-Hikari
[19:43:39 INFO]:  Sellwand, SkinsRestorer, Spawn, StackMob, StaffSpectate, SuperVanish, SynkLibs, TAB, TabTPS, Tebex
[19:43:39 INFO]:  TrollV4, Vault, ViaBackwards, ViaVersion, VoidWorldGenerator, VotifierPlus, VotingPlugin, Vulcan, WildStacker, WorldGuard

### Paper version

version
[19:43:53 INFO]: Checking version, please wait...
[19:43:53 INFO]: This server is running Paper version 1.21.4-232-ver/1.21.4@12d8fe0 (2025-06-09T10:15:42Z) (Implementing API version 1.21.4-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.21.7-16-6e598f8 (MC: 1.21.7)

### Plugin version

about AntiRedstoneClock-Remastered
[19:44:10 INFO]: AntiRedstoneClock-Remastered version 2.2.4
[19:44:10 INFO]: Authors: OneLiteFeather and TheMeinerLP

### Other

_No response_

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
@@ -72,25 +72,21 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
     public void onLoad() {
         saveDefaultConfig();
         reloadConfig();
-        // WorldGuard flag registration is handled during DI initialization
-    }
-
-    @Override
-    public void onEnable() {
-        // Initialize Guice injector
         injector = Guice.createInjector(
             new ServiceModule(this),
             new ExternalSupportModule(this),
             new CommandModule(),
             new ListenerModule()
         );
-        
-        // Get services from injector
+        this.worldGuardSupport = injector.getInstance(WorldGuardSupport.class);
+    }
+
+    @Override
+    public void onEnable() {
         TranslationService translationService = injector.getInstance(TranslationService.class);
         this.redstoneClockService = injector.getInstance(RedstoneClockService.class);
         this.updateService = injector.getInstance(UpdateService.class);
         this.tps = injector.getInstance(CheckTPS.class);
-        this.worldGuardSupport = injector.getInstance(WorldGuardSupport.class);
         this.plotsquaredSupport = injector.getInstance(PlotsquaredSupport.class);
         
         // Setup translations
@@ -225,11 +221,6 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
                 getServer().getPluginManager().registerEvents(listener2, this);
             }
         }
-    }
-
-    private void enableRedstoneClockService() {
-        // RedstoneClockService now provided via dependency injection
-        // using the interface-based architecture from the factory
     }
 
     private void enableTPSChecker() {

